File: java/org/gem/engine/io/reader/definition/Point.java
/*
 * [COPYRIGHT]
 *
 * [NAME] is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

package org.gem.engine.io.reader.definition;

/**
 * Describes a simple matrix bidimensional point.
 * 
 * @author Andrea Cerisara
 * @version $Id: Point.java 537 2010-06-16 18:29:36Z acerisara $
 */
public class Point
{

    private final int row;
    private final int column;

    /**
     * @param row the row of the point
     * @param column the column of the point
     */
    public Point(int row, int column)
    {
        this.row = row;
        this.column = column;
    }

    /**
     * Returns the row of this point.
     * 
     * @return the row of this point
     */
    public int getRow()
    {
        return row;
    }

    /**
     * Returns the column of this point.
     * 
     * @return the column of this point
     */
    public int getColumn()
    {
        return column;
    }

}

File: openquake/hazardlib/scalerel/point.py
# -*- coding: utf-8 -*-
# vim: tabstop=4 shiftwidth=4 softtabstop=4
#
# Copyright (C) 2013-2025 GEM Foundation
#
# OpenQuake is free software: you can redistribute it and/or modify it
# under the terms of the GNU Affero General Public License as published
# by the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# OpenQuake is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.

"""
Module :mod:`openquake.hazardlib.scalerel.point` implements :class:`PointMSR`.
"""
import numpy
from openquake.hazardlib.scalerel.base import BaseMSR


class PointMSR(BaseMSR):
    """
    Implements magnitude-area scaling relationship to mimic point ruptures.
    Independently of the magnitude value, this scaling relationship returns
    always a very small value (1e-4 squared km, corresponding to a 10 by 10 m
    square) for the median area.

    NOTE: This scaling-relationship is meant to be used in area and point
    sources to mimic point ruptures. Is not intended to be used in fault
    sources, as it would require a fault surface discretization step to small
    (less than 10 m, using an aspect ratio equal to 1) which is unfeasible for
    realistic applications.
    """
    def get_median_area(self, mag, rake):
        """
        Returns a value equal to 1e-4 squared km independently of ``mag`` and
        ``rake`` values.

        >>> point_msr = PointMSR()
        >>> 1e-4 == point_msr.get_median_area(4.0, 50)
        True
        >>> 1e-4 == point_msr.get_median_area(9.0, 0)
        True
        """
        return numpy.full_like(mag, 1e-4)
